python/python/tests/test_pubsub.py

```diff
@@ -0,0 +1,1800 @@
+# Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0
+
+from __future__ import annotations
+
+import asyncio
+from enum import IntEnum
+from typing import Any, Dict, List, Optional, Set, Tuple, Union
+
+import pytest
+from glide.async_commands.core import InfoSection
+from glide.config import (
+    ClusterClientConfiguration,
+    ProtocolVersion,
+    RedisClientConfiguration,
+)
+from glide.constants import OK
+from glide.exceptions import WrongConfiguration
+from glide.redis_client import RedisClient, RedisClusterClient, TRedisClient
+from tests.conftest import create_client
+from tests.utils.utils import check_if_server_version_lt, get_random_string
+
+
+class PubSubChannelModes(IntEnum):
+    Exact = 0
```

Comment:
Why not to use `PubSubChannelModes`?

Reply:
I want the test to look nicer, and also pass mypy.
look at `create_channels_and_patterns`
if not that I would need to check the clsuter_mode for every test and also have this long deceleration